pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BaseTableUpsertMetadataManager.java

```diff
@@ -162,12 +162,14 @@ private void preloadSegments()
     LOGGER.info("Preload segments from table: {} for fast upsert metadata recovery", _tableNameWithType);
     onPreloadStart();
     IdealState idealState = HelixHelper.getTableIdealState(_helixManager, _tableNameWithType);
+    Preconditions.checkState(idealState != null, "Failed to find ideal state for table: %s", _tableNameWithType);
```

Comment:
hmm.. NPE didn't get printed by the warn log at L137 above?

Reply:
It will print NPE without exception message, which is not as explicit as `Failed to find ideal state...`